aidants_connect_habilitation/admin.py

```diff
@@ -84,11 +132,115 @@ def accept_request(self, request, queryset):
+        email_body = loader.render_to_string(
+            "email/demande_acceptee.txt", {"organisation": object}
+        )
+        email_subject = (
+            "Aidants Connect - la demande d'habilitation n° "
+            f"{object.data_pass_id} a été acceptée"
+        )
+        initial = {
+            "email_subject": email_subject,
+            "email_body": email_body,
+        }
+        view_context = {
+            **self.admin_site.each_context(request),
+            "media": self.media,
+            "object_id": object_id,
+            "object": object,
+            "form": AdminAcceptationForm(object, initial=initial),
+        }
+
+        return render(
+            request,
+            "aidants_connect_habilitation/admin/organisation_request/accept_form.html",
+            view_context,
+        )
+
+    def __accept_request_post(self, request, object_id):
+        object = OrganisationRequest.objects.get(id=object_id)
+        form = AdminAcceptationForm(object, data=request.POST)
+        if not form.is_valid():
+            return HttpResponseNotAllowed()
+
+        object.accept_request_and_create_organisation()
+        subject = form.cleaned_data.get("email_subject")
+        body_text = form.cleaned_data.get("email_body")
+        self.send_acceptance_email(object, body_text, subject)
+
+        aidant_count = object.aidant_requests.count()
+        if object.manager.is_aidant:
+            aidant_count += 1
+        self.message_user(
+            request,
+            (
+                f"Tout s'est bien passé. La demande {object.data_pass_id} a "
+                f"été acceptée. L'organisation {object.name} et le compte aidant "
+                f"du responsable ont été créés. Les {aidant_count} aidants à former "
+                "nécessaires ont été créés également."
+            ),
+        )
+
+        return HttpResponseRedirect(
+            reverse(
+                "otpadmin:aidants_connect_habilitation_organisationrequest_changelist"
+            )
+        )
+
+    def send_acceptance_email(self, object, body_text=None, subject=None):
+        if body_text:
+            text_message = body_text
+        else:
+            text_message = loader.render_to_string(
+                "email/demande_acceptee.txt", {"organisation": object}
+            )
+        html_message = loader.render_to_string(
+            "email/empty.html", {"content": mark_safe(linebreaks(text_message))}
+        )
+
+        if subject is None:
+            subject = (
+                "Aidants Connect - la demande d'habilitation n° "
+                f"{object.data_pass_id} a été acceptée"
+            )
+
+        recipients = set(object.aidant_requests.values_list("email", flat=True))
+        recipients.add(object.manager.email)
+        recipients.add(object.issuer.email)
+
+        send_mail(
+            from_email=settings.EMAIL_ORGANISATION_REQUEST_ACCEPTANCE_FROM,
+            recipient_list=list(recipients),
```

Comment:
Je me demande si ça marcherait ici de garder le `set`.

Reply:
Je vois pas de raison valable pour laquelle ça ne fonctionnerai pas avec n'imoprte quel `Iterable` (Python :heart:)

Reply:
alors à priori si , il y a une raison. Parce qu'on va utiliser tout ça pour créer un EmailMessage, qui a une méthode qui s'appelle recipients() et qui fait : 
def recipients(self):
        """
        Return a list of all recipients of the email (includes direct
        addressees as well as Cc and Bcc entries).
        """
        return [email for email in (self.to + self.cc + self.bcc) if email]
        
        
du coup si self.to est un set et self.cc une liste .. ben ca va péter.